code/modules/mob/living/carbon/slime/slime.dm

```diff
@@ -894,6 +894,87 @@ mob/living/carbon/slime/var/temperature_resistance = T0C+75
 	icon_state = "bottle14"
 	w_class = W_CLASS_TINY
 
+/obj/item/weapon/slimechange
+	name = "slime dye"
+	desc = "A bottle of dye that can be used on a slime to change it's color."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "bottle14"
+	w_class = W_CLASS_TINY
+
+/obj/item/weapon/slimechange/attack(var/mob/living/M, mob/user as mob)
+	if(istype(M, /mob/living/carbon/slime))
+		var/mob/living/carbon/slime/OS = M
+		if(OS.stat)
+			to_chat(user, "<span class='warning'>You're unable to use the [src] on the dead slime!</span>")
+			return..()
+		var/new_color = input(user, "Choose a new color for the dye:", "Color Select") as anything in list(
+			//"adamantine",
+			//"black",
+			"blue",
+			//"bluespace",
+			//"cerulean",
+			"darkpurple",
+			"darkblue",
+			"darkpurple",
+			"gold",
+			"grey",
+			//"lightpink",
+			"metal",
+			//"oil",
+			"orange",
+			"pink",
+			"purple",
+			//"pyrite",
+			"red",
+			//"sepia",
+			"silver",
+			"yellow")
+		if(new_color && !OS.stat)
+			var/adult = FALSE //Passinging on if it was an adult or not
+			if(istype(OS, /mob/living/carbon/slime/adult))
+				adult = TRUE
+			var/new_slime
+			if(new_color == "grey")
+				new_slime = /mob/living/carbon/slime
+			else
+				new_slime = text2path("/mob/living/carbon/slime[adult ? "/adult" : ""]/[new_color]")
+			var/mob/living/carbon/slime/NS = new new_slime(OS.loc)
+			//Passing on several other traits to make it almost exactly the same as the old one.
+			NS.Friends = OS.Friends
+			NS.tame = OS.tame
+			NS.Discipline = OS.Discipline
+			NS.cores = OS.cores
+			NS.powerlevel = OS.powerlevel
+			NS.amount_grown = OS.amount_grown
+			NS.nutrition = OS.nutrition
+			if(OS.mind)
+				OS.mind.transfer_to(NS)
+			to_chat(user, "<span class='notice'>You dye the slime [new_color].</span>")
+			qdel(OS)
+			qdel(src)
+		else
+			if(OS.stat)
+				to_chat(user, "<span class='warning'>You're unable to use the [src] on the dead slime!</span>")
+			else
+				to_chat(user, "<span class='notice'>You decide not to use the potion.</span>")
+	else if(ishuman(M))
+		var/mob/living/carbon/human/S = M
+		if(isslimeperson(S))
+			var/new_color = input(user, "Choose a color for the dye:", "Color Select") as color|null
+			if(new_color)
```

Comment:
All sorts of things could have happened between this and the line above, and you're checking for none of them. For instance: The two mobs might not be adjacent anymore, the user might not exist anymore, be dead or incapacitated, the target might be dead or not exist anymore, the slime dye itself could have been used up, and so on.

Reply:
Same as above except the simplemob part replaced with other people.